sourcegraph/sourcegraph
# Search: enable `repo:has.topic()` for GitLab

add support to `has.topic` for gitlab topics. Previously only worked for github topics. 
## Test plan
- Confirm new query does not break github topics
- Confirm new index works for gitlab topics
- Run EXPLAIN query locally to confirm `idx_repo_gitlab_topics` is being used
```
sourcegraph=# EXPLAIN SELECT *
FROM repo
WHERE external_service_type = 'gitlab'
AND metadata->'topics' @> '["test-topic"]';
                                                        QUERY PLAN
--------------------------------------------------------------------------------------------------------------------------
 Bitmap Heap Scan on repo  (cost=12.69..300.47 rows=89 width=687)
   Recheck Cond: (((metadata -> 'topics'::text) @> '["test-topic"]'::jsonb) AND (external_service_type = 'gitlab'::text))
   ->  Bitmap Index Scan on idx_repo_gitlab_topics  (cost=0.00..12.67 rows=89 width=0)
         Index Cond: ((metadata -> 'topics'::text) @> '["test-topic"]'::jsonb)
(4 rows)
```

## Preview 🤩
[Preview Link](https://docs.sourcegraph.com/@jhh/search-gitlab-topics)
